core/trino-spi/src/main/java/io/trino/spi/block/DecimalAggregationAccumulatorBlock.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.block;
```

Comment:
Please move the block to `trino-main`.
Extract decimal serde classes, states and block as a separate `aggregation.state` package (e.g `aggregation.state.decimal`)